sympy/integrals/manualintegrate.py

```diff
@@ -1578,9 +1578,8 @@ def manualintegrate(f, var):
     sympy.integrals.integrals.Integral.doit
     sympy.integrals.integrals.Integral
     """
-    if len(f.args) > 1 :
-        if all(i.has(exp) for i in f.args):
-            f = cancel(f)
+    if all(i.has(exp) for i in f.args):
```

Comment:
This looks rather special. Is this cancelling necessary?

Reply:
When an expression is of type `exp*exp` in sympy it is not combining by itself like -
```
In [6]: (exp(x)+exp(2*x))*exp(2*x)                                              
Out[6]: 
⎛ 2⋅x    x⎞  2⋅x
⎝ℯ    + ℯ ⎠⋅ℯ   
```
Due to this such type of integrals taking a large time(example of issue takes about 15 min) so I just combine these into single expression such that now it has to do only `exp` type of integral.